src/trace/xray-service.ts

```diff
@@ -187,13 +198,30 @@ export class XrayService {
     if (lastPart.length !== 24) return;
 
     // We want to turn the last 63 bits into a decimal number in a string representation
-    // Unfortunately, all numbers in javascript are represented by float64 bit numbers, which
-    // means we can't parse 64 bit integers accurately.
-    const hex = new BigNumber(lastPart, 16);
-    if (hex.isNaN()) return;
-
-    // Toggle off the 64th bit
-    const last63Bits = hex.mod(new BigNumber("8000000000000000", 16));
-    return last63Bits.toString(10);
+    try {
+      return (BigInt("0x" + lastPart) % BigInt("0x8000000000000000")).toString(10); // mod by 2^63 will leave us with the last 63 bits
+    } catch (_) {
+      logDebug(`Faied to convert trace id ${lastPart}`);
+      return undefined;
+    }
+  }
+
+  public static extraceDDContextFromAWSTraceHeader(amznTraceId: string): DatadogTraceHeaders | null {
+    const awsContext = XrayService.parseAWSTraceHeader(amznTraceId);
+    if (!awsContext) {
```

Comment:
```suggestion
    if (awsContext === undefined) {
```

Sorry, I kept seeing `something === undefined` and here we are switching it, to `!something`, this is mostly a nit, but I think we should keep consistency on this specific conditional 🤔